src/core/public/chrome/ui/header/recent_items.tsx

```diff
@@ -213,45 +225,37 @@ export const RecentItems = ({
         type: item.meta?.type || '',
         id: item.id,
       }));
-
     if (savedObjects.length) {
       bulkGetDetail(savedObjects, http).then((res) => {
-        const filteredNavLinks = navLinks.filter((link) => !link.hidden);
-        const formatDetailedSavedObjects = res.map((obj) => {
+        const formatDetailedSavedObjects = res.flatMap((obj) => {
```

Comment:
can we do filtering by workspace before bulkGet as `ChromeRecentlyAccessedHistoryItem` has workspace already.